website/static/js/folderPickerNodeConfig.js

```diff
@@ -0,0 +1,438 @@
+/**
+ * Abstract module that controls the addon node settings. Includes Knockout view-model
+ * for syncing data, and HGrid-folderpicker for selecting a folder.
+ */
+'use strict';
+
+var ko = require('knockout');
+require('knockout.punches');
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var FolderPicker = require('js/folderpicker');
+var ZeroClipboard = require('zeroclipboard');
+ZeroClipboard.config('/static/vendor/bower_components/zeroclipboard/dist/ZeroClipboard.swf');
+var $osf = require('js/osfHelpers');
+
+var oop = require('js/oop');
+
+ko.punches.enableAll();
+
+var FolderPickerViewModel = oop.defclass({
+    /*
+     @class ViewModel
```

Comment:
Also, can you add documentation that clarifies how this class is meant to be used? Are there any required overrides? Common overrides?